File: 1025/py/1025.py
# dp

class Solution:
    def divisorGame(self, n: int) -> bool:
        if n == 1:
            return False

        # make dp list
        dp = [False for _ in range(n+1)]

        # dp[1] = False

        # go through, Alice go first
        for i in range(2, n+1):
            for j in range(1, i):
                if i % j == 0:
                    if dp[i-j] == False:
                        dp[i] = True
                        break

        return dp[n]

File: 1025/js/1025.js
/**
 * @param {number} n
 * @return {boolean}
 */
var divisorGame = function(n) {
    if (n === 1){
        return false;
    }

    // make dp
    let dp = new Array(n+1).fill(false);

    // dp[i] means must win or loss at i
    // dp[1] = false

    // go through, Alice go first
    for(let i = 2; i <= n; i++){
        for(let j = 1; j < i; j++){
            if (i % j === 0){
                if (dp[i-j] === false){
                    dp[i] = true;
                    break;
                }
            }
        }
    }

    return dp[n];
};